pkg/providers/aws/provider_blobstorage.go

```diff
@@ -550,3 +554,55 @@ func (p *BlobStorageProvider) getS3BucketConfig(ctx context.Context, bs *v1alpha
 func buildEndUserCredentialsNameFromBucket(b string) string {
 	return fmt.Sprintf("cro-aws-s3-%s-creds", b)
 }
+
+func buildBlobStorageInfoMetricLabels(cr *v1alpha1.BlobStorage, clusterID, bucketName string) map[string]string {
```

Comment:
Given that a lot of these functions are being replicated across all of the aws providers and snapshot controllers, we could possibly extract them into some util functions? Just a thought, it might be overkill. Wdyt?

Reply:
Your right its a little dry at the moment. Could move it all to the metrics util. Currently this is not working as expected as the status are not returning correctly.

Reply:
After a second look it may be too closely coupled to the individual providers and controllers, so when decoupling a lot of the logic would end up living in the provider and controllers.